### PR TITLE
Add support for the Jiran

### PR DESCRIPTION
**Pull request by krikun98**
The Jiran is an open-source keyboard very popular in Russia. Plenty of people have been using this shield since March.
https://github.com/Ladniy/jiran
I can squash all the commits if necessary.

## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [x] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] `.conf` file has optional extra features commented out

### CODE REVIEW

File: app/boards/shields/jiran/jiran.dtsi [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	default_transform: keymap_transform_0 {
+		compatible = "zmk,matrix-transform";
+		columns = <12>;
+		rows = <5>;
+
+			// | SW1  | SW2  | SW3  | SW4  | SW5  | SW6  |   | SW6  | SW5  | SW4  | SW3  | SW2  | SW1  |
+			// | SW7  | SW8  | SW9  | SW10 | SW11 | SW12 |   | SW12 | SW11 | SW10 | SW9  | SW8  | SW7  |
+			// | SW13 | SW14 | SW15 | SW16 | SW17 | SW18 |   | SW18 | SW17 | SW16 | SW15 | SW14 | SW13 |
+			//                      | SW19 | SW20 | SW21 |   | SW21 | SW20 | SW19 |
```
Comment (Nicell) [inline app/boards/shields/jiran/jiran.dtsi:23]: It looks like this is the wrong shape compared to the actual transform. This has 42 keys whereas the map has 56. The same goes for the other ones. Also I'm thinking the maps should be aligned using spaces, as the tabs result in something I assume wasn't intended on GitHub

**Review by Nicell — CHANGES_REQUESTED** [review]
One note on the transforms, and then the `jiran_left` and `jiran_right` `.conf`s should be left out.

File: app/boards/shields/jiran/jiran.dtsi (hunk 1 above)
Comment (krikun98) [reply app/boards/shields/jiran/jiran.dtsi:23]: Oh, I'll fix that up in a second, sorry.

**Comment by krikun98** [comment]
> One note on the transforms, and then the `jiran_left` and `jiran_right` `.conf`s should be left out.

So the local builds use the unified config now? Great!

File: app/boards/shields/jiran/jiran.dtsi (hunk 1 above)
Comment (krikun98) [reply app/boards/shields/jiran/jiran.dtsi]: I initially copied these in when I was editing the shield from the Jian back in March, totally forgot to fix them up for the PR. Done!

File: app/boards/shields/jiran/jiran.dtsi [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+					map = <
+            RC(0,0) RC(0,1) RC(0,2) RC(0,3) RC(0,4) RC(0,5)  RC(0,6) RC(0,7) RC(0,8) RC(0,9) RC(0,10) RC(0,11)
+    RC(4,0) RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4) RC(1,5)  RC(1,6) RC(1,7) RC(1,8) RC(1,9) RC(1,10) RC(1,11) RC(4,11)
+            RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(2,5)  RC(2,6) RC(2,7) RC(2,8) RC(2,9) RC(2,10) RC(2,11)
+            RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4) RC(3,5)  RC(3,6) RC(3,7) RC(3,8) RC(3,9) RC(3,10) RC(3,11)
+                                    RC(4,3) RC(4,4) RC(4,5)  RC(4,6) RC(4,7) RC(4,8)
+		>;
+	};
+
+		jian_transform: keymap_transform_1 {
```
Comment (Nicell) [inline app/boards/shields/jiran/jiran.dtsi:34]:
```suggestion
	jian_transform: keymap_transform_1 {
```

File: app/boards/shields/jiran/jiran.dtsi [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+//                             | SW25 | SW26 | SW27 |   | SW27 | SW26 | SW25 |
+        map = <
+    RC(4,0) RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4) RC(1,5)  RC(1,6) RC(1,7) RC(1,8) RC(1,9) RC(1,10) RC(1,11) RC(4,11)
+            RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(2,5)  RC(2,6) RC(2,7) RC(2,8) RC(2,9) RC(2,10) RC(2,11)
+            RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4) RC(3,5)  RC(3,6) RC(3,7) RC(3,8) RC(3,9) RC(3,10) RC(3,11)
+                                    RC(4,3) RC(4,4) RC(4,5)  RC(4,6) RC(4,7) RC(4,8)
+        >;
+    };
+
+		crkbd_transform: keymap_transform_2 {
```
Comment (Nicell) [inline app/boards/shields/jiran/jiran.dtsi:51]:
```suggestion
	crkbd_transform: keymap_transform_2 {
```

File: app/boards/shields/jiran/jiran.dtsi [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		jian_transform: keymap_transform_1 {
+		compatible = "zmk,matrix-transform";
+		columns = <12>;
+		rows = <5>;
+
+// | SW0  | SW7  | SW8  | SW9  | SW10 | SW11 | SW12 |   | SW12 | SW11 | SW10 | SW9  | SW8  | SW7  | SW0  |
+//        | SW13 | SW14 | SW15 | SW16 | SW17 | SW18 |   | SW18 | SW17 | SW16 | SW15 | SW14 | SW13 |
+//        | SW19 | SW20 | SW21 | SW22 | SW23 | SW24 |   | SW24 | SW23 | SW22 | SW21 | SW20 | SW19 |
+//                             | SW25 | SW26 | SW27 |   | SW27 | SW26 | SW25 |
+        map = <
```
Comment (Nicell) [inline app/boards/shields/jiran/jiran.dtsi:43]:
```suggestion
		map = <
```

File: app/boards/shields/jiran/jiran.dtsi [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		compatible = "zmk,matrix-transform";
+		columns = <12>;
+		rows = <5>;
+
+//        | SW1  | SW2  | SW3  | SW4  | SW5  | SW6  |   | SW6  | SW5  | SW4  | SW3  | SW2  | SW1  |
+// | SW0  | SW7  | SW8  | SW9  | SW10 | SW11 | SW12 |   | SW12 | SW11 | SW10 | SW9  | SW8  | SW7  | SW0  |
+//        | SW13 | SW14 | SW15 | SW16 | SW17 | SW18 |   | SW18 | SW17 | SW16 | SW15 | SW14 | SW13 |
+//        | SW19 | SW20 | SW21 | SW22 | SW23 | SW24 |   | SW24 | SW23 | SW22 | SW21 | SW20 | SW19 |
+//                             | SW25 | SW26 | SW27 |   | SW27 | SW26 | SW25 |
+					map = <
```
Comment (Nicell) [inline app/boards/shields/jiran/jiran.dtsi:25]:
```suggestion
		map = <
```

File: app/boards/shields/jiran/jiran.dtsi [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+//        | SW13 | SW14 | SW15 | SW16 | SW17 | SW18 |   | SW18 | SW17 | SW16 | SW15 | SW14 | SW13 |
+//        | SW19 | SW20 | SW21 | SW22 | SW23 | SW24 |   | SW24 | SW23 | SW22 | SW21 | SW20 | SW19 |
+//                             | SW25 | SW26 | SW27 |   | SW27 | SW26 | SW25 |
+        map = <
+    RC(4,0) RC(1,0) RC(1,1) RC(1,2) RC(1,3) RC(1,4) RC(1,5)  RC(1,6) RC(1,7) RC(1,8) RC(1,9) RC(1,10) RC(1,11) RC(4,11)
+            RC(2,0) RC(2,1) RC(2,2) RC(2,3) RC(2,4) RC(2,5)  RC(2,6) RC(2,7) RC(2,8) RC(2,9) RC(2,10) RC(2,11)
+            RC(3,0) RC(3,1) RC(3,2) RC(3,3) RC(3,4) RC(3,5)  RC(3,6) RC(3,7) RC(3,8) RC(3,9) RC(3,10) RC(3,11)
+                                    RC(4,3) RC(4,4) RC(4,5)  RC(4,6) RC(4,7) RC(4,8)
+        >;
+    };
```
Comment (Nicell) [inline app/boards/shields/jiran/jiran.dtsi:49]:
```suggestion
		>;
	};
```

**Review by Nicell — COMMENTED** [review]
A bunch of small formatting things, otherwise the transforms look great now, thanks!

**Pull request merged**